### PR TITLE
Adding java8 version of Concurrent

### PR DESCRIPTION
**Pull request by mswatosh**
Starting this as a draft, just to show the idea and run a full build to see additional implications of the change. This will need more commits before a full review including but not limited to:

- [x] Updating additional concurrent features
- [ ] Handling NLS between the `concurrent` and `concurrent.java8` projects
- [ ] Cleanup bnd files

### CODE REVIEW

**Comment by mswatosh** [comment]
#build

**Comment by mswatosh** [comment]
#build

**Comment by mswatosh** [comment]
#build

**Comment by posmith** [comment]
L2 message review completed.

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1000.unrecognized.property.explanation=An unrecognized property is configured on a context service.
+CWWKC1000.unrecognized.property.useraction=Remove or correct the unrecognized property.
+
+# Moved from com.ibm.ws.context
+CWWKC1001.context.providers.unavailable=CWWKC1001E: The configured thread context {0} is unavailable for the configuration element {1}.
+CWWKC1001.context.providers.unavailable.explanation=The application server is unable to find provider(s) of the specified type(s) of thread context.
+CWWKC1001.context.providers.unavailable.useraction=Ensure the provider(s) of the requested thread context are enabled.
+
+# Moved from com.ibm.ws.context
+CWWKC1002.provider.cardinality.violation=CWWKC1002E: The configuration element {0} cannot have more than one sub-element of type {1}.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:40]:
```suggestion
CWWKC1002.provider.cardinality.violation=CWWKC1002E: The configuration element {0} cannot have more than one subelement of type {1}.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1002.provider.cardinality.violation.explanation=At most, one element of each type of context provider may be configured under a context service.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:41]:
```suggestion
CWWKC1002.provider.cardinality.violation.explanation=At most, one element of each type of context provider can be configured under a context service.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1002.provider.cardinality.violation.useraction=Remove duplicate context provider elements from the context service configuration.
+
+# Moved from com.ibm.ws.context
+CWWKC1020.baseContextRef.infinite=CWWKC1020E: The attribute baseContextRef of configuration element {0} causes an infinite chain of dependencies.
+CWWKC1020.baseContextRef.infinite.explanation=An infinite chain of dependencies exists because baseContextRef directly or indirectly points back to the same context service instance.
+CWWKC1020.baseContextRef.infinite.useraction=Update the context service configuration to break the infinite chain of dependencies.
+
+CWWKC1100.resource.unavailable=CWWKC1100E: The resource {0} is unavailable.
+CWWKC1100.resource.unavailable.explanation=A resource might become unavailable due to the server shutting down or because of a configuration change, either to itself or another service upon which it depends.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:50]:
```suggestion
CWWKC1100.resource.unavailable.explanation=A resource can become inaccessible if the server shuts down or due to a change in configuration, affecting either the resource itself or another dependent service.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1002.provider.cardinality.violation=CWWKC1002E: The configuration element {0} cannot have more than one sub-element of type {1}.
+CWWKC1002.provider.cardinality.violation.explanation=At most, one element of each type of context provider may be configured under a context service.
+CWWKC1002.provider.cardinality.violation.useraction=Remove duplicate context provider elements from the context service configuration.
+
+# Moved from com.ibm.ws.context
+CWWKC1020.baseContextRef.infinite=CWWKC1020E: The attribute baseContextRef of configuration element {0} causes an infinite chain of dependencies.
+CWWKC1020.baseContextRef.infinite.explanation=An infinite chain of dependencies exists because baseContextRef directly or indirectly points back to the same context service instance.
+CWWKC1020.baseContextRef.infinite.useraction=Update the context service configuration to break the infinite chain of dependencies.
+
+CWWKC1100.resource.unavailable=CWWKC1100E: The resource {0} is unavailable.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:49]:
```suggestion
CWWKC1100.resource.unavailable=CWWKC1100E: The resource {0} is inaccessible.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1100.resource.unavailable.explanation=A resource might become unavailable due to the server shutting down or because of a configuration change, either to itself or another service upon which it depends.
+CWWKC1100.resource.unavailable.useraction=Verify the configuration of the resource and any resources upon which it depends.
+
+CWWKC1110.task.canceled=CWWKC1110I: The task {0}, which was submitted to executor service {1}, is canceled.
+CWWKC1110.task.canceled.explanation=Execution of the task has been canceled.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:54]:
```suggestion
CWWKC1110.task.canceled.explanation=The task execution is now canceled.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1110.task.canceled.useraction=No action is required.
+
+CWWKC1101.task.failed=CWWKC1101E: The task {0}, which was submitted to executor service {1}, failed with the following error: {2}.
+CWWKC1101.task.failed.explanation=A task submitted to an executor has failed. Refer to the exception message for the cause.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:58]:
```suggestion
CWWKC1101.task.failed.explanation=A task submitted to an executor fails. Refer to the exception message for the cause..
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1101.task.failed.useraction=Correct the cause of the failure and resubmit the task.
+
+CWWKC1102.listener.failed=CWWKC1102E: The managed task listener for task {0}, which was submitted to executor service {1}, failed with the following error: {2}.
+CWWKC1102.listener.failed.explanation=A managed task listener operation has failed. Refer to the exception message for the cause.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:62]:
```suggestion
CWWKC1102.listener.failed.explanation=A managed task listener operation fails. Refer to the exception message for the cause.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1102.listener.failed.useraction=Correct the cause of the failure and decide whether or not to resubmit the task.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:63]:
```suggestion
CWWKC1102.listener.failed.useraction=Correct the cause of the failure and decide whether to resubmit the task.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+CWWKC1103.skip.run.failed=CWWKC1103E: Execution of task {0}, which was submitted to executor service {1}, is skipped because the Trigger.skipRun operation failed with the following error: {2}.
+CWWKC1103.skip.run.failed.explanation=The skipRun operation for the Trigger has failed, causing the current execution attempt for the task to be skipped.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:66]:
```suggestion
CWWKC1103.skip.run.failed.explanation=The skipRun operation for the trigger fails, causing the current execution attempt for the task to be skipped.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1103.skip.run.failed.useraction=Correct the cause of the failure and decide whether or not to resubmit the task.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:67]:
```suggestion
CWWKC1103.skip.run.failed.useraction=Correct the cause of the failure and decide whether to resubmit the task.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+CWWKC1111.task.invalid=CWWKC1111E: The task {0} is not valid.
+CWWKC1111.task.invalid.explanation=The task cannot be submitted because it is not valid.
+CWWKC1111.task.invalid.useraction=Update the application to submit a valid task to the ExecutorService.
+
+CWWKC1112.all.tasks.canceled=CWWKC1112I: All tasks are canceled.
+CWWKC1112.all.tasks.canceled.explanation=All of the tasks were canceled before any task could be invoked.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:74]:
```suggestion
CWWKC1112.all.tasks.canceled.explanation=All tasks were canceled before any task could be invoked.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+CWWKC1112.all.tasks.canceled.useraction=No action is required.
+
+CWWKC1120.future.get.rejected=CWWKC1120E: The Future supplied to ManagedTaskListener methods taskSubmitted and taskStarting cannot be used to wait for task completion.
+CWWKC1120.future.get.rejected.explanation=The ManagedTaskListener taskSubmitted and taskStarting methods run on the thread that will submit or start the task. Therefore, the Future supplied to these methods must not be used to wait for the task.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:78]:
```suggestion
CWWKC1120.future.get.rejected.explanation=The ManagedTaskListener taskSubmitted and taskStarting methods run on the thread that submits or starts the task. Therefore, the Future supplied to these methods must not be used to wait for the task.
```

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops (hunk 6 above)
Comment (helyarp) [reply dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:58]: "executor failed" ("failed", not "has failed" or "fails")

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops (hunk 7 above)
Comment (helyarp) [reply dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:62]: "operation failed" ("failed", not "has failed" or "fails")

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops (hunk 9 above)
Comment (helyarp) [reply dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:66]: "trigger failed" ("failed", not "has failed" or "fails")

File: dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops (hunk 11 above)
Comment (helyarp) [reply dev/com.ibm.ws.concurrent.java8/resources/com/ibm/ws/concurrent/resources/CWWKCMessages.nlsprops:74]: Suggestion: Replace "could be" with "was"

**Pull request closed without merging**